compiler/rustc_mir_transform/src/cross_crate_inline.rs

```diff
@@ -135,7 +135,12 @@ impl<'tcx> Visitor<'tcx> for CostChecker<'_, 'tcx> {
                     }
                 }
             }
-            TerminatorKind::Call { unwind, .. } => {
+            TerminatorKind::Call { ref func, unwind, .. } => {
+                if let Some((fn_def_id, _)) = func.const_fn_def() {
+                    if self.tcx.intrinsic(fn_def_id).is_some() {
```

Comment:
Nit: this would benefit from combining into one `if` using either let-chaining or `is_some_and`.